### PR TITLE
CAMEL-10292 - camel-zookeeper added new RoutePolicy

### PR DESCRIPTION
**Pull request by paoloantinori**

### CODE REVIEW

**Comment by oscerd** [comment]
Looks good to me, can you please run mvn -Psourcecheck and fix the codestyle? Thanks @paoloantinori ! :-)

**Comment by oscerd** [comment]
Thanks the PR has been merged :-)

Do you mind closing this? and maybe you can add some information in the .adoc file for the zookeeper component :-)

Thanks!

**Comment by paoloantinori** [comment]
I'll add something to https://github.com/apache/camel/blob/master/components/camel-zookeeper/src/main/docs/zookeeper-component.adoc , thank you!

File: components/camel-zookeeper/src/main/java/org/apache/camel/component/zookeeper/ZooKeeperProducer.java [hunk 1]
```diff
@@ -64,6 +64,9 @@ public ZooKeeperProducer(ZooKeeperEndpoint endpoint) {
 
     public void process(Exchange exchange) throws Exception {
 
+        if (connection == null) {
```
Comment (alaendle) [inline components/camel-zookeeper/src/main/java/org/apache/camel/component/zookeeper/ZooKeeperProducer.java]: Could the change to ZooKeeperProducer.java be merged to v2.18.x?
(see also https://issues.apache.org/jira/browse/CAMEL-10961)

**Comment by oscerd** [comment]
What we can try to do is backporting the whole commit.

**Comment by alaendle** [comment]
As far as I can see the change in ZooKeeperProducer is really independent from the rest; or did I miss something? So while I wouldn't argue against backporting the added features using curator, I didn't think it's necessary to fix the issue 10961. Please correct me if I'm wrong!

**Pull request closed without merging**